pulp_ansible/app/tasks/collections.py

```diff
@@ -596,6 +596,28 @@ async def _post_save(self, batch):
                 :class:`~pulpcore.plugin.stages.DeclarativeContent` objects to be saved.
 
         """
+
+        def _get_file_obj(tar, file_path, artifact_path):
```

Comment:
@awcrosby I'd like you to take a look at this. It seems like opening the tarball and extracting the manifest should be something that happens during the import instead of as a separate function that pulp performs when a collection is added.

The one snag here is that this would also require the sync logic to be updated to pull down the manifest and files list.

What do you think?

Reply:
when `pulp-ansible` calls `galaxy-importer` the `MANIFEST.json` is extracted and validated, as long as this step occurs after that validation then it should be ok

it looks like pulp-ansible was already untaring in `_post_save()` and this PR makes it so data gets saved to the `CollectionVersion`

Reply:
yes, this is what is blocking us from doing on-demand sync, we always open the tarball during sync,
the importer is only used when uploading collections.